motor_control/twoWheelSpeed.h

```diff
@@ -0,0 +1,57 @@
+#ifndef TWOWHEELSPEED_H
+#define TWOWHEELSPEED_H
+
+//Arbitrary values for now
+#define KP 0.05
+#define KD 0.05
+
+void twoWheelSpeed(double wheelRatio, int throttleRead, int* leftPWM, int* rightPWM, byte currRightSpeed, byte currLeftSpeed, byte* prevError, bool leftFast, Ramp* ramp)
+{
+    byte currFastSpeed;
+    byte currSlowSpeed;
+    byte fastSpeedToWrite;
+    int slowSpeedToWrite;
+    byte error;
+    
+    if(leftFast)
+    {
+        currFastSpeed = currLeftSpeed;
+        currSlowSpeed = currRightSpeed;
+        
+        fastSpeedToWrite = *leftPWM;
+        slowSpeedToWrite = *rightPWM;
+    }
+    else
+    {
+        currFastSpeed = currRightSpeed;
+        currSlowSpeed = currLeftSpeed;
+        
+        fastSpeedToWrite = *rightPWM;
+        slowSpeedToWrite = *leftPWM;
+    }
+    
+    //use ramping to get desired fast wheel speed
+    fastSpeedToWrite = ramp->newSpd(throttleRead, millis());
+    
+    //calculate desired slow wheel speed based on ratio and fast wheel speed
+    //use PD to adjust slow wheel speed
+    error = currSlowSpeed - currFastSpeed * wheelRatio;
+    slowSpeedToWrite += KP*error + KD*(error - *prevError);
+    slowSpeedToWrite = max(0, min(slowSpeedToWrite, 255));
+    
+    *prevError = error;
+    
+    //update PWM values
+    if(leftFast)
+    {
+        *leftPWM = fastSpeedToWrite;
+        *rightPWM = slowSpeedToWrite;
+    }
+    else
+    {
+        *rightPWM = fastSpeedToWrite;
+        *leftPWM = slowSpeedToWrite;
+    }
```

Comment:
Pointers could get pretty messy and annoying, especially if they're null at the start of the program. I'm guessing that they're in use here because we need to return multiple results from this function, so maybe consider defining a `struct` that you could take in and return? Completely your call though, pointers will work just as well functionally :)